util/reggen/gen_dv.py

```diff
@@ -85,6 +143,8 @@ def gen_dv(block: IpBlock, dv_base_prefix: str, outdir: str) -> int:
     generated = []
 
     lblock = block.name.lower()
+    block_dv_base_names = get_block_base_name(dv_base_names, lblock)
```

Comment:
I am not sure I understand this. Wouldn't it be better to pass the dict generated by `get_dv_base_names_objects` to the template on line 163 below?

Reply:
Even otherwise, I think the template needs to know the raw list of overrides. So you should be passing `dv_base_names` directly, rather than `block_dv_base_names`.

Reply:
Oh I see, but I believe the `reggen/uvm_reg.sv.tpl` template assumes to generate RAL for one register block (https://github.com/lowRISC/opentitan/blob/master/util/reggen/uvm_reg.sv.tpl#L14). So I think there we can just pass the specific dv_base_names for that block instead of passing the entire list.

Reply:
i thought since the script only take into account of one block, so why not parse the information first and just input the relevant information about the base_class names?

Reply:
Sorry my mistake - i understand this a little better now. I will still recommend calling `get_dv_base_names_objects` outside of `get_block_base_name`.

The reggen template (which looks at one block) needs a single `DvBaseNames` object. What you are doing here is fine, but could be refactored to:
```python
    dv_base_names_map = get_dv_base_names_objects(dv_base_names)
    block_dv_base_names = get_block_base_name(dv_base_names_map, lblock)
```
However, for the topgen UVM ral template, you are calling `get_block_base_name` in a for loop for each sub-block (see line 32 of `top_uvm_reg_sv.tpl`), which will compute the same map again and again. By refactoring this out, you compute the map only once outside the for loop and provide the generated map as input to `get_block_base_name`.

Reply:
I see your point, thanks for the detailed explanation. I agree that it will be more optimized for topgen :) Update the code for both, Thanks again for all the suggestions.